server/src/main/java/io/crate/execution/ddl/tables/AlterTableClient.java

```diff
@@ -180,6 +182,13 @@ private CompletableFuture<Long> setSettings(BoundAlterTable analysis) {
                 analysis.excludePartitions(),
                 analysis.settings()
             );
+            GenericProperties<Object> withProperties = analysis.withProperties();
+            Object timeout = withProperties.get("timeout");
+            if (timeout != null) {
```

Comment:
Ended up adding the WITH timeout support for `setSetting` too because it makes the docs simpler. Stating that WITH (timeout) only works for `SET (number_of_shards...)` felt a bit clunky.